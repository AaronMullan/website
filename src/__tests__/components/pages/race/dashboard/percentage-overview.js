import React from 'react'
import renderer from 'react-test-renderer'
import PercentageOverview from '~components/pages/race/dashboard/percentage-overview'

describe('Components : Race : Dashboard : Percentage Overview', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <PercentageOverview
          stateName={'New Mexico'}
          dataType={'race'}
          casePercent={0.22}
          deathPercent={0.043}
          className={'race-class'}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()

    const nullTree = renderer
      .create(
        <PercentageOverview
          stateName={'New Mexico'}
          dataType={'race'}
          className={'race-class'}
        />,
      )
      .toJSON()
    expect(nullTree).toMatchSnapshot()

    const nullDeathTree = renderer
      .create(
        <PercentageOverview
          stateName={'New Mexico'}
          dataType={'race'}
          casePercent={0.22}
          className={'race-class'}
        />,
      )
      .toJSON()
    expect(nullDeathTree).toMatchSnapshot()
  })
})
