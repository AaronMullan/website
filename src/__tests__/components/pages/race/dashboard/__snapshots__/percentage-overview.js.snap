// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Race : Dashboard : Percentage Overview renders correctly 1`] = `
<div
  className="race-class"
>
  <h3
    className="title"
  >
    Race
     completeness for 
    New Mexico
    :
  </h3>
  <div
    className="data"
  >
    <div
      className="dataItem"
    >
      <p>
        <span
          className="percent"
        >
          78
          %
        </span>
         
        <span
          className="a11y-only"
        >
          of
        </span>
         
        <span
          className="percentCaption"
        >
          Cases have no 
          race
           data
        </span>
      </p>
    </div>
    <div
      className="dataItem"
    >
      <p>
        <span
          className="percent"
        >
          96
          %
        </span>
         
        <span
          className="a11y-only"
        >
          of
        </span>
         
        <span
          className="percentCaption"
        >
          Deaths have no 
          race
           data
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`Components : Race : Dashboard : Percentage Overview renders correctly 2`] = `null`;

exports[`Components : Race : Dashboard : Percentage Overview renders correctly 3`] = `
<div
  className="race-class"
>
  <h3
    className="title"
  >
    Race
     completeness for 
    New Mexico
    :
  </h3>
  <div
    className="data"
  >
    <div
      className="dataItem"
    >
      <p>
        <span
          className="percent"
        >
          78
          %
        </span>
         
        <span
          className="a11y-only"
        >
          of
        </span>
         
        <span
          className="percentCaption"
        >
          Cases have no 
          race
           data
        </span>
      </p>
    </div>
    <div
      className="dataItem"
    >
      <p>
        <span
          className="percent"
        >
          <img
            alt="Alert icon"
            src="test-file-stub"
          />
        </span>
         
        <span
          className="a11y-only"
        >
          of
        </span>
         
        <span
          className="percentCaption"
        >
          Deaths have no 
          race
           data
        </span>
      </p>
    </div>
  </div>
</div>
`;
